drivers/timer/cortex_m_systick.c

```diff
@@ -43,6 +43,13 @@ static uint32_t last_load;
 #define cycle_t uint32_t
 #endif
 
+/* Build time or runtime assert depending on
+ * CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME.
```

Comment:
The only driver that supports that runtime weirdness (which is a historical wart we should really remove) is HPET, this seems like a noop here?

Reply:
Currently that config is also used for SiLabs Gecko BURTC timer, Renesas RA4M1 SoC and a few other places I haven't looked into.

Renesas RA4M1 is a Cortex-M SoC that uses this systick driver. It would need to be modified to not use `CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME` before this could be a build time only assert.

I **_think_** the reason those drivers use `CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME` is because they specify the system clock prescaler in the device tree (e.g. commit where Renesas RA4M1 added clock control https://github.com/zephyrproject-rtos/zephyr/commit/1741b3a356fadfb01f1375b08cadb7b8e395bfb4).

I noticed stm32 had a reasonable workaround for that:

```
config SYS_CLOCK_TICKS_PER_SEC
	default 4096 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" < 16
	default 2048 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 16
	default 1024 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 32
	default  512 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 64
	default  256 if "$(dt_node_int_prop_int,$(DT_STM32_LPTIM_PATH),$(DT_ST_PRESCALER))" = 128
	depends on STM32_LPTIM_TIMER && STM32_LPTIM_CLOCK_LSE
```

Which seems nicer than having to calculate all timeouts at runtime.

What are your thoughts on this?